tests/trial_tests/test_fixed_frozen.py

```diff
@@ -0,0 +1,253 @@
+import datetime
+from typing import Any
+from typing import Dict
+
+import pytest
+
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalDistribution
+from optuna.distributions import FloatDistribution
+from optuna.distributions import IntDistribution
+from optuna.trial import BaseTrial
+from optuna.trial import create_trial
+from optuna.trial import FixedTrial
+from optuna.trial import FrozenTrial
+
+
+parametrize_trial_type = pytest.mark.parametrize("trial_type", [FixedTrial, FrozenTrial])
+
+
+def _create_trial(
+    trial_type: type,
+    params: Dict[str, Any] = {"x": 10},
+    distributions: Dict[str, BaseDistribution] = {"x": FloatDistribution(5, 12)},
+) -> BaseTrial:
+    if trial_type == FixedTrial:
+        return FixedTrial(params)
+    elif trial_type == FrozenTrial:
+        trial = create_trial(value=0.2, params=params, distributions=distributions)
+        trial.number = 0
+        return trial
+    else:
+        assert False
+
+
+@parametrize_trial_type
+def test_suggest_float(trial_type: type) -> None:
```

Comment:
I think we can dry `test_suggest_*` by parametrizing the methods and its arguments.

Reply:
Hmm... Considering the amount of difference in each test function and considering that these tests test deprecated functions, I think it may be a little over-engineering to parametrize `test_suggest_*`.  I think we could reconsider it when `suggest_uniform` etc. are removed.

Reply:
There is a subtle difference between `suggest_float` and `suggest_int` about the accepted value of `(step, log)` that prevents it from being DRYed.

Reply:
Sounds great. It would be fixed after removing `suggest_uniform` and etc.